### PR TITLE
KSZ9477 errata fixes

### PR DESCRIPTION
**Pull request by jpaali**
Includes a couples of fixes from the official errata.
And in separate commit custom fixes which are needed to get the camera to work.
Also removed unnecessary phyinit from mpfs_ethernet..

### CODE REVIEW

File: drivers/net/ksz9477.c [hunk 1]
```diff
@@ -211,6 +219,391 @@ static int ksz9477_sgmii_write_indirect(uint32_t address, uint16_t *value,
...
+  ret = ksz9477_reg_write16(KSZ9477_PHY_REMOTE_LP(port), 0x00f0);
+  if (ret != OK)
+    {
+      nerr("PHY Remote Loopback register write failure, ret %d\n", ret);
+      return ret ? ret : -EINVAL;
+    }
+
+  /* Needed configurations for MMD Signal Quality */
+
+  err1[0].reg = 0xce;
```
Comment (jlaitine) [inline drivers/net/ksz9477.c:376]: These could be directly initialized at the array definition; like "const struct...err1 = { {...}, {...}...};"

File: drivers/net/ksz9477.c [hunk 2]
```diff
@@ -211,6 +219,391 @@ static int ksz9477_sgmii_write_indirect(uint32_t address, uint16_t *value,
...
+      ret = ksz9477_reg_write16(KSZ9477_PHY_MMD_DATA(port), *value);
+      value++;
+    }
+
+out:
+
+  return ret;
+}
+
+#if 1
```
Comment (jlaitine) [inline drivers/net/ksz9477.c:344]: Maybe leave only what is actially used and remove extra "#if 1" / "#if 0" or at least comment why the dead code is left in?

Comment (jpaali) [reply drivers/net/ksz9477.c:344]: I was pondering that these "official" erratas could be upstreamed..

Comment (jpaali) [reply drivers/net/ksz9477.c]: We decided not to upstream these. Drop this. Backup in backup_ksz9477_errata if needed at some point..

File: drivers/net/ksz9477.c (hunk 1 above)
Comment (jpaali) [reply drivers/net/ksz9477.c]: done

**Review by maseabunikie — APPROVED** [review]

**Comment by jpaali** [comment]
We decided not to upstream these. Dropping "official" errata implementation.. Backup in backup_ksz9477_errata if needed at some point..

**Pull request merged**